lib/transaction_repository.rb

```diff
@@ -55,6 +55,12 @@ def transaction_success_checker(invoice_id)
     end.delete_if { |k, v| !k.include?(true) }.values.flatten
   end
 
+  def transaction_failure_checker(invoice_id)
+    invoice_id.group_by do |invoice_id|
+      find_all_by_attribute("invoice_id", invoice_id).flat_map { |t| t.result == "success" }
+    end.delete_if { |k, v| k.include?(true) }.values.flatten
```

Comment:
Unused block argument - `v`. If it's necessary, use `_` or `_v` as an argument name to indicate that it won't be used.<br>Avoid multi-line chains of blocks.